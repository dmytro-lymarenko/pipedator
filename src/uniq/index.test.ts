import { uniq } from './index';

describe('uniq()', () => {
	it('should validate when value is uniq', () => {
		expect(uniq().validate([1, 2, 3])).toBeNull();
	});

	it('should validate with custom comparator', () => {
		expect(uniq((a, b) => a.x === b.x).validate([{ x: 1 }, { x: 2 }, { x: 3 }])).toBeNull();
	});

	it('should validate when there is one or 0 items in any direction', () => {
		expect(uniq().validate([])).toBeNull();
		expect(uniq().validate([1])).toBeNull();
	});

	it('should fail when array is not uniq', () => {
		expect(uniq().validate([1, 2, 4, 2])).toMatchSnapshot();
		expect(uniq().validate([1, 1, 1, 5, 4])).toMatchSnapshot();
		expect(uniq().validate([1, 2, 3, 5, 4, 6, 7, 1])).toMatchSnapshot();
	});

	it('should fail when array is not uniq with custom comparator', () => {
		expect(uniq((a, b) => a.x === b.x).validate([{ x: 2 }, { x: 2 }, { x: 3 }])).toMatchSnapshot();
	});

	it('should support custom message as first arg', () => {
		expect(uniq('Custom message').validate([5, 1, 3, 1, 2])).toMatchSnapshot();
	});

	it('should support both custom comparator and custom message', () => {
		expect(uniq((a, b) => a.x === b.x, 'Custom message').validate([{ x: 2 }, { x: 2 }, { x: 3 }])).toMatchSnapshot();
	});
});
