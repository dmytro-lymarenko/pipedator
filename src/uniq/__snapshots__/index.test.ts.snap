// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uniq() should fail when array is not uniq 1`] = `
Object {
  "children": null,
  "message": "Array should contain uniq items",
  "path": "",
}
`;

exports[`uniq() should fail when array is not uniq 2`] = `
Object {
  "children": null,
  "message": "Array should contain uniq items",
  "path": "",
}
`;

exports[`uniq() should fail when array is not uniq 3`] = `
Object {
  "children": null,
  "message": "Array should contain uniq items",
  "path": "",
}
`;

exports[`uniq() should fail when array is not uniq with custom comparator 1`] = `
Object {
  "children": null,
  "message": "Array should contain uniq items",
  "path": "",
}
`;

exports[`uniq() should support both custom comparator and custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;

exports[`uniq() should support custom message as first arg 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
